src/main/java/randoop/test/ContractCheckingVisitor.java

```
@@ -99,6 +101,71 @@ public TestChecks visit(ExecutableSequence s) {
+          // Create three ExecutionOutcome objects from the ExecutableSequence
+          ExecutionOutcome result1 = s.getResult(i);
+          ExecutionOutcome result2 = s.getResult(j);
+          ExecutionOutcome result3 = s.getResult(k);
+
+          if (Log.isLoggingOn()) {
+            Log.logLine("Checking contract " + c.getClass() + " on " + i + ", " + j + ", " + k);
+          }
+
+          ExecutionOutcome exprOutcome =
+                  ObjectContractUtils.execute(
+                          c,
+                          ((NormalExecution) result1).getRuntimeValue(),
+                          ((NormalExecution) result2).getRuntimeValue(),
+                          ((NormalExecution) result3).getRuntimeValue());
+
+          if (exprOutcome instanceof NormalExecution) {
+            NormalExecution e = (NormalExecution) exprOutcome;
+            if (e.getRuntimeValue().equals(true)) {
+              if (Log.isLoggingOn()) Log.logLine("Contract returned true.");
+              // Behavior ok.
+            } else {
+              if (Log.isLoggingOn())
+                Log.logLine("Contract returned false. Will add ExpressionEqFalse check");
+              // Create an check that records the actual value
+              // returned by the expression, marking it as invalid
+              // behavior.
+              checks.add(new ObjectCheck(c, i, s.sequence.getVariable(i), s.sequence.getVariable(j), s.sequence.getVariable(k)));
+            }
+          } else if (exprOutcome instanceof ExceptionalExecution) {
+            Throwable e = ((ExceptionalExecution) exprOutcome).getException();
+            if (Log.isLoggingOn()) Log.logLine("Contract threw exception: " + e.getMessage());
+            if (e instanceof BugInRandoopException) {
+              throw new BugInRandoopException(e);
+            }
+            // Execution of contract resulted in exception. Do not create
+            // a contract-violation decoration.
```

I don't think this is  just when `e.getRuntimeValue().equals(true)` evaluates to false, but I'm not sure.

I'm trying to understand under what circumstances execution reaches line 156, which seems to be whenever an exception other than `BugInRandoopException` is thrown.  The code seems to silently ignore all such exceptions.  

Ignoring an exception is a code smell and I wonder what is the justification for ignoring the exception.